packages/application-extension/src/index.ts

```diff
@@ -3,13 +3,131 @@
 
 import { JupyterFrontEndPlugin, JupyterFrontEnd } from '@jupyterlab/application';
 
+import { ICommandPalette, Dialog, showDialog } from '@jupyterlab/apputils';
+
+import { IDocumentManager } from '@jupyterlab/docmanager';
+
+import {
+  IDocumentProvider,
+  IDocumentProviderFactory,
+  ProviderMock
+} from '@jupyterlab/docprovider';
+
+import { WebsocketProvider } from 'y-websocket';
+
+import { IMainMenu } from '@jupyterlab/mainmenu';
+
 import { ITranslator, TranslationManager } from '@jupyterlab/translation';
 
+const YJS_WEBSOCKET_URL = 'wss://demos.yjs.dev';
+
+class WebSocketProvider extends WebsocketProvider implements IDocumentProvider {
+  constructor(options: IDocumentProviderFactory.IOptions) {
+    super(YJS_WEBSOCKET_URL, options.guid, options.ymodel.ydoc);
```

Comment:
For now this is passed the path to the file:

https://github.com/jupyterlab/jupyterlab/blob/abcb9242af1a97fa9e897d34f0556e07bb913fa3/packages/docregistry/src/context.ts#L86

Maybe we can add another plugin that would create a uuid, that could then be shared with other folks via a query string parameter, and be concatenated to the `guid`.

Reply:
Or use the jupyterlab-video chat approach: https://github.com/jtpio/jupyterlite/issues/88#issuecomment-841518843